### PR TITLE
feat: Invoke migration batch process on click of the migration button.

### PR DESCRIPTION
**Pull request by vrundakansara**
### Description
Task - [Link](https://trello.com/c/YEFxfhDw/930-migrate-on-migration-click)

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests

### CODE REVIEW

File: inc/core/builder/class-astra-builder-admin.php [hunk 1]
```diff
@@ -110,9 +110,13 @@ public function migrate_to_builder() {
 			wp_send_json_error( __( 'You don\'t have the access', 'astra' ) );
 		}
 
-		$migrate = isset( $_POST['value'] ) ? sanitize_key( $_POST['value'] ) : '';
-		$migrate = ( $migrate ) ? true : false;
+		$migrate        = isset( $_POST['value'] ) ? sanitize_key( $_POST['value'] ) : '';
+		$migrate        = ( $migrate ) ? true : false;
+		$migration_flag = get_option( 'astra-v3-option-migration', false );
```
Comment (premanshup) [inline inc/core/builder/class-astra-builder-admin.php:115]: @vrundakansara Please use `astra_get_option` here.

File: inc/theme-update/astra-update-functions.php [hunk 2]
```diff
@@ -452,6 +450,8 @@ function astra_header_builder_migration() {
 	update_option( 'astra-settings', $theme_options );
 	update_option( 'sidebars_widgets', $widget_options );
```
Comment (premanshup) [inline inc/theme-update/astra-update-functions.php:451]: @vrundakansara Please use `astra-settings` to add value.

File: inc/theme-update/astra-update-functions.php [hunk 3]
```diff
@@ -452,6 +450,8 @@ function astra_header_builder_migration() {
...
 
+	update_option( 'astra-v3-option-migration', true );
```
Comment (premanshup) [inline inc/theme-update/astra-update-functions.php:453]: @vrundakansara Please use `astra-settings` to add flags.

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/core/builder/class-astra-builder-admin.php (hunk 1 above)
Comment (vrundakansara) [reply inc/core/builder/class-astra-builder-admin.php]: Yes, thanks. Done.

File: inc/theme-update/astra-update-functions.php (hunk 3 above)
Comment (vrundakansara) [reply inc/theme-update/astra-update-functions.php]: Yes, thanks. Done.

File: inc/theme-update/astra-update-functions.php (hunk 2 above)
Comment (vrundakansara) [reply inc/theme-update/astra-update-functions.php:453]: Yes, thanks. Done.

**Review by rohitp-bsf — APPROVED** [review]

**Pull request merged**